mage_ai/data_preparation/repo_manager.py

```diff
@@ -212,20 +235,20 @@ def get_variables_dir(repo_path: str = None) -> str:
     return get_repo_config(repo_path=repo_path).variables_dir
 
 
-config = get_repo_config()
 project_uuid = None
 try:
-    project_uuid = config.project_uuid
+    with get_repo_config().metadata_path as f:
```

Comment:
It's a bit confusing here. Why can't we use `config.project_uuid`?

Reply:
@wangxiaoyou1993 the problem is that the `RepoConfig` is read before the server is finished starting. If a user tries to use a secret variable in the project metadata, getting the `RepoConfig` will fail and `config.project_uuid` will return `None` because the db connection session may not be initialized yet.

Reply:
The `RepoConfig` being able to use the template vars causes a lot of these dependency issues.

Reply:
It doesn't directly resolve that circular dependency, and I didn't see any errors for that dependency. It would probalby be a good idea to fix that though, so I'll look into it